### PR TITLE
feat(leanpkg): add initial support for test command

### PR DESCRIPTION
**Pull request by jroesch**
Making a few tweaks to `leanpkg` added a simple test command which reuses the LEAN_PATH and invokes `--make` in the test directory. You can see an example on the `smt2_interface` repository.

### CODE REVIEW

**Comment by jroesch** [comment]
Leo and I talked about a simple hack to implement a testing framework which reports the number of failures and successes. We can simply generate `@[test] def _test_framework.fresh_name_XXX : bool := (tt | ff)` each indicating a success or failure of the test, allowing us to generate a nice report summary. Going to work up a demo of that this afternoon.

**Comment by jroesch** [comment]
@Kha @gebner do you guys have more thoughts on how we should do this? trying to move things out of the standard library and being able to test packages are important to this.

**Comment by gebner** [comment]
I think the general approach is good.  It is simple and pretty future-proof in that we do not commit to support any testing framework, but only promise to execute all files in a given directory.  IIRC, Haskell has the same convention.

I don't have strong opinions on directory layout, but if we start having `test` directories etc., maybe we should really push to have `src` directories (and enable them by default).

> We can simply generate @[test] def _test_framework.fresh_name_XXX : bool := (tt | ff)

Another solution that is pretty popular is to annotate your tests with `@[test]` and then have the test runner look up all definitions in the environment that have the `test` attribute and execute them.  Then we don't need to store anything in the environment.

File: library/init/meta/tactic.lean [hunk 1]
```diff
@@ -86,6 +86,13 @@ meta def fail_if_success {α : Type u} (t : tactic α) : tactic unit :=
  (λ a s, mk_exception "fail_if_success combinator failed, given tactic succeeded" none s)
  (λ e ref s', success () s)
 
+meta def success_if_fail {α : Type u} (t : tactic α) : tactic unit :=
```
Comment (gebner) [inline library/init/meta/tactic.lean:89]: This change is completely unrelated, right?

File: leanpkg/leanpkg/main.lean [hunk 2]
```diff
@@ -43,13 +43,26 @@ d ← read_manifest,
 io.put_str_ln $ "configuring " ++ d.name ++ " " ++ d.version,
 assg ← solve_deps d,
 path_file_cnts ← mk_path_file <$> construct_path assg,
+io.put_str_ln path_file_cnts,
```
Comment (gebner) [inline leanpkg/leanpkg/main.lean]: This is just left-over debugging code, right?  If you actually want to print it, I think we should output it with some kind of description.

File: library/init/meta/tactic.lean (hunk 1 above)
Comment (jroesch) [reply library/init/meta/tactic.lean:89]: Yes, this is useful for testing tactic failures, and I am using this branch to update & test smt2, it seemed easy to add while making testing related changes.

File: leanpkg/leanpkg/main.lean (hunk 2 above)
Comment (jroesch) [reply leanpkg/leanpkg/main.lean]: I removed this, probably just forgot to push the cleanup commit.
